chain/deals/provider_api.go

```diff
@@ -0,0 +1,109 @@
+package deals
```

Comment:
I'm ok with leaving this off for now, be eventually we need to find a way to unit test the provider code, including any new code like this.